src/reconcile/rec_visibility.c

```diff
@@ -118,9 +118,8 @@ __rec_append_orig_value(
      * delete a value respectively at timestamp 0 and 10, and later insert it again at 20. We need
      * the tombstone to tell us there is no value between 10 and 20.
      */
-    if (unpack->tw.stop_ts != WT_TS_MAX || unpack->tw.stop_txn != WT_TXN_MAX) {
-        tombstone_globally_visible =
-          __wt_txn_visible_all(session, unpack->tw.stop_txn, unpack->tw.durable_stop_ts);
+    if (__wt_time_window_has_stop(&unpack->tw)) {
+        tombstone_globally_visible = __wt_txn_tw_stop_visible_all(session, &unpack->tw);
```

Comment:
Here is the problem, we didn't take into the prepare flag into the consideration of global visibility.